assets/js/base/context/providers/cart-checkout/checkout-state/index.tsx

```diff
@@ -311,11 +315,16 @@ export const CheckoutStateProvider = ( {
 					if ( successResponse && ! errorResponse ) {
 						dispatch( actions.setComplete( successResponse ) );
 					} else if ( isObject( errorResponse ) ) {
-						if ( errorResponse.message ) {
-							const errorOptions = errorResponse.messageContext
-								? { context: errorResponse.messageContext }
-								: undefined;
-							addErrorNotice(
+						if (
+							errorResponse.message &&
+							isString( errorResponse.message )
+						) {
+							const errorOptions =
+								errorResponse.messageContext &&
+								isString( errorResponse.messageContext )
+									? { context: errorResponse.messageContext }
```

Comment:
Because the `errorResponse` in the code above is given a type of `Record< string, unknown > | null` and the `Response` gets assigned to it, the one below is operating directly on the `Response`.